torch/lib/c10d/reducer.cpp

```diff
@@ -1498,16 +1526,18 @@ void Reducer::set_parameter_stats() {
   }
 }
 
-void Reducer::set_bucket_stats() {
+std::string Reducer::get_bucket_stats() {
+  std::string bucket_sizes = "";
   for (const auto& bucket : buckets_) {
     const auto& variables = bucket.replicas[0].variables;
     int bucket_size = 0;
     for (const auto& v : variables) {
       bucket_size += v.numel() * v.element_size();
     }
-    ddp_logging_data_->bucket_sizes += std::to_string(bucket_size);
-    ddp_logging_data_->bucket_sizes += ", ";
+    bucket_sizes += std::to_string(bucket_size);
```

Comment:
String concatenation in a loop is inefficient, especially if there are many buckets.

Better to construct a vector with a reserved capacity by `reserve`, and then casts it into a string when it needs to be displayed. I think we can use `folly::join` for more efficient string concatenation.

Reply:
comment as above, will change it as vector<int>. meanwhile pytorch does not support to use folly though

Reply:
You can also use `std::ostringstream` for casting the vector to a string for efficiency.